File: lecture_005/vector_findAll.cpp
#include<iostream>
#include<vector>
using namespace std;

void display(vector<int>& aans)
{
    for(int i:aans)
    cout<<i<<" ";

}

vector<int> find(vector<int>& arr,int data)
{
    vector <int> ans;
    for(int i=0;i<arr.size();i++)
    {
        if(arr[i]==data)
        ans.push_back(i);
    }
    return ans;
}

void insertValue(vector<int>& arr)
{
    for(int i=0;i<arr.size();i++)
    cin>>arr[i];
}

void solve()
{
    cout<<"enter n"<<endl;
    int n;
    cin>>n;
    vector<int> arr(n,0);
    insertValue(arr);
    cout<<"enter data index to find"<<endl;
    int b;
    cin>>b;
    vector<int>aans=find(arr,b);
    display(aans);
}

int main(int args, char** argv)
{
    solve();
    return 0;
}

File: lecture_005/vector_findAll.java
import java.util.Scanner;
import java.util.ArrayList;
public class vector_findAll{
    public static Scanner scn=new Scanner(System.in);
    public static void main(String[] args)
    {
        solve();
        return;
    }
    public static void solve()
    {
        System.out.println("enter n");
        int n=scn.nextInt();
        int[]arr=new int[n];
        inputArray(arr);
        System.out.println("enter data");
        int data=scn.nextInt();
        ArrayList<Integer> ans=allIndex(arr,data);
        System.out.println(ans);
    }
    public static void inputArray(int[]arr)
    {
        for(int i=0;i<arr.length;i++)
            arr[i]=scn.nextInt();
        System.out.println();
    }
    public static ArrayList<Integer> allIndex(int[] arr,int b)
    {
        ArrayList<Integer>ans=new ArrayList<>();
        for(int i=0;i<arr.length;i++)
        {
            if(arr[i]==b)
            ans.add(i);
        }
        return ans;
    }
    
}